runtimevar/httpvar/httpvar.go

```diff
@@ -36,6 +44,53 @@ import (
 	"time"
 )
 
+func init() {
+	o := &URLOpener{Client: http.DefaultClient}
+	for _, scheme := range Schemes {
+		runtimevar.DefaultURLMux().RegisterVariable(scheme, o)
+	}
+}
+
+// Schemes are the URL schemes httpvar registers its URLOpener under on runtimevar.DefaultMux.
+var Schemes = []string{"http", "https"}
+
+// URLOpener opens HTTP URLs like "http://myserver.com/foo.txt".
+//
+// The full URL, including scheme, is used as the endpoint, except that the
+// the following URL parameters are removed if present:
+//   - decoder: The decoder to use. Defaults to runtimevar.BytesDecoder.
+//       See runtimevar.DecoderByName for supported values.
+type URLOpener struct {
+	// The Client to use; required.
+	Client *http.Client
+
+	// Decoder specifies the decoder to use if one is not specified in the URL.
+	// Defaults to runtimevar.BytesDecoder.
+	Decoder *runtimevar.Decoder
+
+	// Options specifies the options to pass to NewVariable.
+	Options Options
+}
+
+// OpenVariableURL opens a httpvar Variable for u.
+func (o *URLOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	// Clone u because we may strip some query parameters.
+	u2 := *u
+	q := u2.Query()
+
+	decoderName := q.Get("decoder")
+	q.Del("decoder")
+	decoder, err := runtimevar.DecoderByName(ctx, decoderName, o.Decoder)
+	if err != nil {
+		return nil, fmt.Errorf("open variable %v: invalid decoder: %v", u, err)
+	}
+	// See if we changed the query parameters.
```

Comment:
Should it also check for invalid parameters?

Reply:
No, for this one I think we should pass them through.